extensions/mssql/src/errorDiagnostics/errorDiagnosticsProvider.ts

```diff
@@ -51,13 +51,20 @@ export class ErrorDiagnosticsProvider extends SqlOpsFeature<any> {
 			}
 
 			protected override registerProvider(options: any): Disposable {
-				let handleConnectionError = async (errorCode: number, errorMessage: string, connection: azdata.connection.ConnectionProfile): Promise<azdata.diagnostics.ConnectionDiagnosticsResult> => {
-					let restoredProfile = this.convertToIConnectionProfile(connection);
-					if (errorCode === ErrorDiagnosticsConstants.MssqlPasswordResetErrorCode) {
-						logDebug(`Error Code ${errorCode} requires user to change their password, launching change password dialog.`)
-						return await this.handleChangePassword(restoredProfile);
+				let handleConnectionError = async (errorInfo: azdata.diagnostics.IErrorInformation, connection: azdata.connection.ConnectionProfile): Promise<azdata.diagnostics.ConnectionDiagnosticsResult> => {
+					if (errorInfo) {
+						let restoredProfile = this.convertToIConnectionProfile(connection);
+
+						if (errorInfo.errorCode === ErrorDiagnosticsConstants.MssqlPasswordResetErrorCode) {
+							logDebug(`ErrorDiagnosticsProvider: Error Code ${errorInfo.errorCode} requires user to change their password, launching change password dialog.`);
+							return await this.handleChangePassword(restoredProfile);
+						}
+						else if (errorInfo.errorCode === ErrorDiagnosticsConstants.MssqlCertValidationFailedErrorCode) {
```

Comment:
Would suggest looking into adding telemetry for this stuff too so you know how much people are using it and whether it succeeds in helping (probably best to do as a follow up task)